src/anyio/_core/_compat.py

```diff
@@ -132,11 +132,12 @@ def _unwrap(self) -> float:
 
 
 class DeprecatedAwaitableList(List[T]):
-    def __init__(self, *args: T, func: Callable[..., 'DeprecatedAwaitableList']):
-        super().__init__(*args)
+    def __init__(self, iterable: Iterable[T] = (), *,
+                 func: Callable[..., 'DeprecatedAwaitableList']):
+        super().__init__(iterable)
```

Comment:
with `*args` `DeprecatedAwaitableList(1, 2)` failed at runtime and `DeprecatedAwaitableList([1, 2])` failed the type check